gsd/hoomd.py

```diff
@@ -1092,3 +1097,58 @@ def open(name, mode='rb'):
                          schema_version=[1, 4])
 
     return HOOMDTrajectory(gsdfileobj)
+
+def read_log(name, scalar_only=True):
+    """Read scalar values from a hoomd schema GSD file.
+
+    Args:
+        name (str): File name to open.
+
+    Returns:
+        `pandas.DataFrame`
+    """
+    if fl is None:
+        raise RuntimeError("file layer module is not available")
+    if gsd is None:
+        raise RuntimeError("gsd module is not available")
+    if pandas is None:
+        raise RuntimeError("pandas module is not available")
+
+    gsdfileobj = fl.open(name=str(name),
+                         mode='rb',
+                         application='gsd.hoomd ' + gsd.__version__,
+                         schema='hoomd',
+                         schema_version=[1, 4])
+
+    logged_data_names = gsdfileobj.find_matching_chunk_names('log/')
+    # Always log timestep associated with each log entry
+    logged_data_names.insert(0, 'configuration/step')
+    if len(logged_data_names) > 1:
+        logged_data_dict = dict()
+        for log in logged_data_names:
+            if gsdfileobj.chunk_exists(frame=0, name=log):
+                tmp = gsdfileobj.read_chunk(frame=0, name=log)
+                if scalar_only and not tmp.shape[0] == 1:
+                    pass
+                else:
+                    logged_data_dict[log] = numpy.zeros_like(tmp, shape=(gsdfileobj.nframes, *tmp.shape))
+                    logged_data_dict[log][0] = tmp
```

Comment:
```suggestion
                if scalar_only and not tmp.shape[0] == 1:
                    continue
                if tmp.shape[0] == 1:
                    logged_data_dict[log] = numpy.full(tmp[0], shape=(gsdfileobj.nframes,))
                else:
                    logged_data_dict[log] = numpy.tile(tmp, (gsdfileobj.nframes, 1))
```
With the change below.

Reply:
The suggestion for `tile` does not match the behavior of the previous code.
```
>>> a                                                                                     
array([[-1.,  2., -3.],                                                                   
       [ 0.,  2., -4.],              
       [-3.,  2.,  1.],
       [ 1.,  2.,  3.]], dtype=float32)
>>> b = numpy.tile(a, (10,1))                                                              
>>> b.shape
(40, 3)
```
`b` should have a shape of `(10,4,3)`. I pushed a unit test that checks per-particle vector logged quantities.
@b-butler what is the correct numpy function to call to create the desired array?

Reply:
```python
np.tile(a, (gsdfileobj.nframes,) + tuple(1 for _ in a.shape))
```

Reply:
Thanks.